playground/backend/internal/setupers/run_builder_setuper.go

```diff
@@ -0,0 +1,50 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setupers
+
+import (
+	pb "beam.apache.org/playground/backend/internal/api/v1"
+	"beam.apache.org/playground/backend/internal/environment"
+	"beam.apache.org/playground/backend/internal/executors"
+	"beam.apache.org/playground/backend/internal/fs_tool"
+	"beam.apache.org/playground/backend/internal/logger"
+	"fmt"
+	"github.com/google/uuid"
+)
+
+// SetupRunBuilder returns executors.RunBuilder based on sdk
+func SetupRunBuilder(pipelineId uuid.UUID, lc *fs_tool.LifeCycle, workingDir string, sdkEnv *environment.BeamEnvs, compileBuilder *executors.CompileBuilder) (*executors.RunBuilder, error) {
```

Comment:
Do we need a different working dir here? in compile builder, you use workingDir := lc.GetAbsoluteExecutableFilesFolderPath()

Reply:
In `compile builder` uses `executableFilesFolder` (`path/{workingDir}/executable_files/{pipelineId}/src`), and `run builder` uses `workingDir` (`path/{workingDir}`).
I renamed this one in `compile builder`. Now it should be clearer.